app/keepers/keepers.go

```diff
@@ -532,6 +537,16 @@ func (ak *AppKeepers) InitKeepers(
 		ak.IBCKeeper.PortKeeper, ak.AccountKeeper, ak.BankKeeper,
 	)
 
+	ak.RatelimitKeeper = *ratelimitkeeper.NewKeeper(
+		appCodec,
+		runtime.NewKVStoreService(keys[ratelimittypes.StoreKey]),
+		ak.GetSubspace(ratelimittypes.ModuleName),
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		ak.BankKeeper,
+		ak.IBCKeeper.ChannelKeeper,
+		ak.IBCKeeper.ChannelKeeper, // ICS4Wrapper
+	)
+
```

Comment:
please address the TODO in the PFM keeper (L570)

```
ak.PFMRouterKeeper = pfmrouterkeeper.NewKeeper(
		appCodec,
		ak.keys[pfmroutertypes.StoreKey],
		ak.TransferKeeper,
		ak.IBCKeeper.ChannelKeeper,
		ak.BankKeeper,
		ak.IBCFeeKeeper, // TODO replace ICS4Wrapper with RateLimitKeeper when implemented
		appparams.AccGov.String(),
	)
```

Reply:
Also in wasm keeper, the ICS4Wrapper should be the RateLimitKeeper, similar to how [Osmosis has it wired up](https://github.com/osmosis-labs/osmosis/blob/289a53ed24c70ac73111ff3c49d19f3a2d3b77c3/app/keepers/keepers.go#L590)

Reply:
Also I see Osmosis [replaces the ICS4Wrapper with the RateLimitKeeper in the wasm keeper](https://github.com/osmosis-labs/osmosis/blob/289a53ed24c70ac73111ff3c49d19f3a2d3b77c3/app/keepers/keepers.go#L590). However, they have their own implementation of the module, not sure if this will have any effect on our case